pymathics/natlang/linguistic_data.py

```diff
@@ -0,0 +1,350 @@
+        wordnet, language_code = self._load_wordnet(
+            evaluation, self._language_name(evaluation, options)
+        )
+        if not wordnet:
+            return
+
+        py_word = self._parse_word(word)
+        if not py_word:
+            return
+
+        senses = self._senses(py_word, wordnet, language_code)
+        if senses is not None:
+            return ListExpression(*[[String(s) for s in desc] for syn, desc in senses])
+
+    def eval_property(self, word, property, evaluation: Evaluation, options: dict):
+        "WordData[word_, property_String, OptionsPattern[WordData]]"
+        if word.get_head is SymbolStringExpression:
+            if property.get_string_value() == "Lookup":
+                return Expression(SymbolDictionaryLookup, word)
+        elif isinstance(word, String) or word.get_head() is SymbolList:
+            return self._property(
+                word, property.get_string_value(), "ShortRules", evaluation, options
+            )
+
+    def eval_property_form(
+        self, word, property, form, evaluation: Evaluation, options: dict
+    ):
+        "WordData[word_, property_String, form_String, OptionsPattern[WordData]]"
+        if isinstance(word, String) or word.get_head() is SymbolList:
+            return self._property(
+                word,
+                property.value,
+                form.value,
+                evaluation,
+                options,
+            )
+
+
+class WordDefinition(_WordNetBuiltin):
+    """
+    <url>:WMA:
+    https://reference.wolfram.com/language/ref/WordDefinition.html</url>
+
+    <dl>
+      <dt>'WordDefinition[$word$]'
+      <dd>returns a definition of $word$ or Missing["Available"] if $word$ is not known.
+    </dl>
+
+    >> WordDefinition["gram"]
+     = {a metric unit of weight equal to one thousandth of a kilogram}
+    """
+
+    summary_text = "retrieve the definition of a word"
+
+    def eval(self, word, evaluation: Evaluation, options: dict):
+        "WordDefinition[word_String, OptionsPattern[WordDefinition]]"
+        wordnet, language_code = self._load_wordnet(
+            evaluation, self._language_name(evaluation, options)
+        )
+        if wordnet:
+            senses = self._senses(word.value.lower(), wordnet, language_code)
+            if senses:
+                return ListExpression(*[String(syn.definition()) for syn, _ in senses])
+            else:
+                return Expression(SymbolMissing, StringNotAvailable)
+
+
+class WordList(_WordListBuiltin):
+    """
+    <url>:WMA:
+    https://reference.wolfram.com/language/ref/WordList.html</url>
+
+    <dl>
+      <dt>'WordList[]'
+      <dd>returns a list of common words.
+
+      <dt>'WordList[$type$]'
+      <dd>returns a list of common words of type $type$.
+    </dl>
+
+    >> N[Mean[StringLength /@ WordList["Adjective"]], 2]
+     = 9.3
+    """
+
+    summary_text = "retrieve a list of common words"
+
+    def eval(self, evaluation: Evaluation, options: dict):
```

Comment:
For this function, I am seeing

```
RecursionError: maximum recursion depth exceeded while calling a Python object
```

Maybe we comment it out for now?

Reply:
In my machine and in the CI it seems to work. Do you have a test case which raise this error?

Reply:
Did you try the first  Form  `WordList[]`  which is what I believe this _particular_ `eval`  method handles? There is no CI test for this, so unless you ran this explicitly you might not know it fails.

Also, a principle mentioned in the guide says that every builtin should have at least one example so that might lead you to a problem as well if there is one.

Reply:
I tried with `WordList[]//Length` . Now I added this as a doctest.

Reply:
And yes, the CI is still clean.

Reply:
> I tried with `WordList[]//Length` . Now I added this as a doctest.

Ok - good. Details are important, giving these up front saves time. 

Yes, `WordList[]//Length` works for me as well. But `WordList[]` by itself does not and gives me the recursion error. 

This is something we should point out. 

Also 

```
Length[WordList[]] > 10000
```

is not not something that a user is going to be interested in. This feels like something that belongs as a pytest instead. 

See https://wolfram.com/xid/0cq048d2-ov2unu for something that is more akin to a user-oriented example.

Reply:
`WordList[];` does not produce the issue either. So, it seems to be a problem in showing large lists.

Reply:
`WordList[]` also works in my system, but takes time to print the result.